integration/update_field_compat_test.go

```diff
@@ -54,6 +58,86 @@ func TestUpdateFieldCompatInc(t *testing.T) {
 	testUpdateCompat(t, testCases)
 }
 
+func TestUpdateFieldCompatMax(t *testing.T) {
+	t.Parallel()
+	setup.SkipForTigrisWithReason(t, "tigris does not support dynamic types")
+
+	testCases := map[string]updateCompatTestCase{
+		"Int32Lower": {
+			update: bson.D{{"$max", bson.D{{"v", int32(30)}}}},
+		},
+		"Int32Higher": {
+			update: bson.D{{"$max", bson.D{{"v", int32(60)}}}},
+		},
+		"Int32Negative": {
+			update: bson.D{{"$max", bson.D{{"v", int32(-22)}}}},
+		},
+		"Document": {
+			update: bson.D{{"$max", bson.D{{"v", bson.D{{"foo", "bar"}}}}}},
+			skip:   "https://github.com/FerretDB/FerretDB/issues/457",
+		},
+		"EmptyDocument": {
+			update: bson.D{{"$max", bson.D{{"v", bson.D{{}}}}}},
+			skip:   "https://github.com/FerretDB/FerretDB/issues/1000",
+		},
+		"Double": {
+			update: bson.D{{"$max", bson.D{{"v", 54.32}}}},
+		},
+		"DoubleNegative": {
+			update: bson.D{{"$max", bson.D{{"v", -54.32}}}},
+		},
+
+		"MultipleQueries": {
+			update: bson.D{{"$max", bson.D{{"v", int32(39)}, {"a", int32(30)}}}},
+		},
+		"DuplicateQuery": {
+			update: bson.D{{"$max", bson.D{{"v", int32(39)}, {"v", int32(30)}}}},
+			skip:   "Handle duplicates correctly",
+		},
```

Comment:
```
expected: mongo.WriteException(mongo.WriteException{WriteConcernError:(*mongo.WriteConcernError)(nil), WriteErrors:mongo.WriteErrors{mongo.WriteError{Index:0, Code:40, Message:"Updating the path 'v' would create a conflict at 'v'", Details:bson.Raw(nil), Raw:bson.Raw(nil)}}, Labels:[]string(nil), Raw:bson.Raw(nil)})
actual  : mongo.CommandError(mongo.CommandError{Code:0, Message:"connection(127.0.0.1:37525[-4]) socket was unexpectedly closed: EOF", Labels:[]string{"NetworkError", "RetryableWriteError"}, Name:"", Wrapped:topology.ConnectionError{ConnectionID:"127.0.0.1:37525[-4]", Wrapped:(*errors.errorString)(0xc00007a120), init:false, message:"socket was unexpectedly closed"}, Raw:bson.Raw(nil)})
```

Do we have any issue related to that? The closest one I found is #1101.

Reply:
> socket was unexpectedly closed: EOF

The real error is in the log above

Reply:
> The real error is in the log above
```
[document.go:242 bson.(*Document).ReadFrom] bson.Document.ReadFrom: types.ConvertDocument: types.Document.validate: keys and values count mismatch: 1 != 2
```
I still believe that key duplicates are handled incorrectly